src/main/java/com/google/gcp_variant_transforms/library/SchemaGeneratorImpl.java

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Google LLC
+
+package com.google.gcp_variant_transforms.library;
+
+import com.google.api.services.bigquery.model.TableFieldSchema;
+import com.google.api.services.bigquery.model.TableSchema;
+import com.google.common.collect.ImmutableList;
+import htsjdk.variant.vcf.VCFHeader;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+/**
+ * Service to create BigQuery Schema from VCFHeader 
+ */
+public class SchemaGeneratorImpl implements SchemaGenerator {
+  
+  public TableSchema getSchema(VCFHeader vcfHeader){
+    ImmutableList<TableFieldSchema> schemaFields = getFields(vcfHeader);
+    TableSchema schema = new TableSchema()
+        .setFields(schemaFields);
+    return schema;
+  }
+
+  public ImmutableList<TableFieldSchema> getFields(VCFHeader vcfHeader){
+    ImmutableList.Builder<TableFieldSchema> fields = new ImmutableList.Builder<TableFieldSchema>();
+    Integer[] fieldIndices = getFieldIndices(SchemaUtils.constantFieldIndexToNameMap);
+
+    for (Integer fieldIndex : fieldIndices){
+      String fieldName = SchemaUtils.constantFieldIndexToNameMap.get(fieldIndex);
+      TableFieldSchema field;
+      if( fieldName == SchemaUtils.FieldName.CALLS){ // requires generating sub-fields
+        ImmutableList<TableFieldSchema> callsSubFields = createCallSubFields();
+        field = new TableFieldSchema()
+            .setName(fieldName)
+            .setMode(SchemaUtils.constantFieldNameToModeMap.get(fieldName))
+            .setDescription(SchemaUtils.constantFieldNameToDescriptionMap.get(fieldName))
+            .setFields(callsSubFields)
+            .setType(SchemaUtils.constantFieldNameToTypeMap.get(fieldName));
+      }else {
```

Comment:
It seems ALT field also has nested field. I am not sure if it is the final version or are you going to deal with it as the sync meeting mentioned

Reply:
In the next PR, I will add the rest of the Schema fields.